pkgs/applications/editors/android-studio/update.sh

```diff
@@ -0,0 +1,70 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -I nixpkgs=./. -i bash -p python3Packages.yq nix-prefetch-scripts
+
+set -euo pipefail
+
+DEFAULT_NIX="$(realpath "./pkgs/applications/editors/android-studio/default.nix")"
+RELEASES_XML="$(curl -v --silent https://raw.githubusercontent.com/JetBrains/intellij-sdk-docs/main/topics/_generated/android_studio_releases.xml 2>/dev/null)"
+
+# Available channels: Release/Patch (stable), Beta, Canary
+getLatest() {
+    local attribute="$1"
+    local channel="$2"
+    [[ "$channel" = "stable" ]] && channel="release" # Our naming convention is different
+    local result="$(echo "$RELEASES_XML" \
+        | xq -r ".[].item[] | select(.channel == \"${channel^}\") | .${attribute}" \
+        | sort --version-sort \
+        | tail -n 1)"
+
+    if [[ -n "$result" ]]; then
+        echo "$result"
+    else
+        echo "could not find the latest $attribute for $channel"
+        exit 1
+    fi
+}
```

Comment:
```suggestion
#! nix-shell -I nixpkgs=./. -i bash -p jq nix-prefetch-scripts

set -euo pipefail

DEFAULT_NIX="$(realpath "./pkgs/applications/editors/android-studio/default.nix")"
RELEASES_JSON="$(curl --silent -L https://jb.gg/android-studio-releases-list.json)" 

# Available channels: Release/Patch (stable), Beta, Canary
getLatest() {
    local attribute="$1"
    local channel="$2"
    case "$channel" in
        "stable") local select='.channel == "Release" or .channel == "Patch"' ;;
        *) local select=".channel == \"${channel^}\"" ;;
    esac
    local result="$(echo "$RELEASES_JSON" \
        | jq -r ".content.item[] | select(${select}) | [.version, .${attribute}] | join(\" \")" \
        | sort --version-sort \
        | cut -d' ' -f 2- \
        | tail -n 1)"

    if [[ -n "$result" ]]; then
        echo "$result"
    else
        echo "could not find the latest $attribute for $channel"
        exit 1
    fi 
}       
``` 

URL is from https://plugins.jetbrains.com/docs/intellij/android-studio-releases-list.html

`xq` is converting XML to JSON to use `jq` so we can directly use `jq`. 

Also make sure that we always sort by `version` attribute.

Handle `Patch` channel for release in the function directly.